src/containerapp/azext_containerapp/_sdk_models.py

```diff
@@ -0,0 +1,3390 @@
+     and modifiedBy information.
+    :vartype system_data: ~commondefinitions.models.SystemData
+    :ivar etag: Resource Etag.
+    :vartype etag: str
+    """
+
+    _validation = {
+        'id': {'readonly': True},
+        'name': {'readonly': True},
+        'type': {'readonly': True},
+        'system_data': {'readonly': True},
+        'etag': {'readonly': True},
+    }
+
+    _attribute_map = {
+        'id': {'key': 'id', 'type': 'str'},
+        'name': {'key': 'name', 'type': 'str'},
+        'type': {'key': 'type', 'type': 'str'},
+        'system_data': {'key': 'systemData', 'type': 'SystemData'},
+        'etag': {'key': 'etag', 'type': 'str'},
+    }
+
+    def __init__(self, **kwargs):
+        super(AzureEntityResource, self).__init__(**kwargs)
+        self.etag = None
+
+
+class AzureFileProperties(Model):
+    """Azure File Properties.
+
+    :param account_name: Storage account name for azure file.
+    :type account_name: str
+    :param account_key: Storage account key for azure file.
+    :type account_key: str
+    :param access_mode: Access mode for storage. Possible values include:
+     'ReadOnly', 'ReadWrite'
+    :type access_mode: str or ~commondefinitions.models.AccessMode
+    :param share_name: Azure file share name.
+    :type share_name: str
+    """
+
+    _attribute_map = {
+        'account_name': {'key': 'accountName', 'type': 'str'},
+        'account_key': {'key': 'accountKey', 'type': 'str'},
+        'access_mode': {'key': 'accessMode', 'type': 'str'},
+        'share_name': {'key': 'shareName', 'type': 'str'},
+    }
+
+    def __init__(self, **kwargs):
+        super(AzureFileProperties, self).__init__(**kwargs)
+        self.account_name = kwargs.get('account_name', None)
+        self.account_key = kwargs.get('account_key', None)
+        self.access_mode = kwargs.get('access_mode', None)
+        self.share_name = kwargs.get('share_name', None)
+
+
+class AzureStaticWebApp(Model):
+    """The configuration settings of the Azure Static Web Apps provider.
+
+    :param state: <code>Disabled</code> if the Azure Static Web Apps provider
+     should not be enabled despite the set registration; otherwise,
+     <code>Enabled</code>. Possible values include: 'Enabled', 'Disabled'
+    :type state: str or ~commondefinitions.models.IdentityProviderState
+    :param registration: The configuration settings of the Azure Static Web
+     Apps registration.
+    :type registration:
+     ~commondefinitions.models.AzureStaticWebAppRegistration
+    """
+
+    _attribute_map = {
+        'state': {'key': 'state', 'type': 'str'},
+        'registration': {'key': 'registration', 'type': 'AzureStaticWebAppRegistration'},
+    }
+
+    def __init__(self, **kwargs):
+        super(AzureStaticWebApp, self).__init__(**kwargs)
+        self.state = kwargs.get('state', None)
+        self.registration = kwargs.get('registration', None)
+
+
+class AzureStaticWebAppRegistration(Model):
```

Comment:
Do we need this file? If we do, do we need all the models here? remember these files will impact the install perf.

Reply:
Since the YAML of the containerapp can be fluid, I wanted to include all the possible models here for deserialization. Once we move to using Python SDK, this will be removed